packages/controllers/src/snaps/keyring.ts

```diff
@@ -0,0 +1,89 @@
+// TODO
+export interface Namespace {
+  chains: { id: string; name: string }[];
+  methods?: string[];
+  events?: string[];
+}
+
+/**
+ * Finds a keyring snap for each namespace implements at a minimum the requested functionality.
+ *
+ * @param requestedNamespaces - The requested namespaces.
+ * @param snaps - All snaps and their exposed keyring namespaces.
+ * @returns A mapping between namespaces and snap ids.
+ */
+export function findMatchingKeyringSnaps(
+  requestedNamespaces: Record<string, Namespace>,
+  snaps: Record<string, Record<string, Namespace>>,
+) {
+  const snapEntries = Object.entries(snaps);
+  return Object.entries(requestedNamespaces).reduce(
+    (acc, [requestedNamespaceId, currentRequestedNamespace]) => {
+      const matchedSnap = snapEntries.find(([_, namespaces]) => {
```

Comment:
Since we're only using the value here, could this simply be `Object.values`?

```suggestion
  const snapEntries = Object.values(snaps);
  return Object.entries(requestedNamespaces).reduce(
    (acc, [requestedNamespaceId, currentRequestedNamespace]) => {
      const matchedSnap = snapEntries.find((namespaces) => {
```

Reply:
We still need the SnapId, which is the key. So I don't think so? It seems the easiest that `matchedSnap` has that value 🤔